### PR TITLE
main HDK readme improvement

### PR DESCRIPTION
**Pull request by Connoropolous**
Since this is one of the first things that is linked to from the bottom of the "install holochain" pages, it will be a big improvement to have these references up to date.
https://developer.holochain.org/install/
Screen Shot 2021-05-05 at 11 29 10 AM

- [x] fix main hdk rust code samples
- [x] move detailed wasm stuff into advanced "ON-WASM.md" and linked to it
- [x] updated hdk function name references
- [x] I didn't change the text in ON-WASM.md at all, so if it's out of date, that won't change here. You can go straight to the heart of the changes here: https://github.com/holochain/holochain/pull/758/files#diff-677db3179240e72af11668c24b04df41e03bc1f962badafc3708623735f8aed4

### CODE REVIEW

File: crates/hdk/ON-WASM.md [hunk 1]
```diff
@@ -0,0 +1,551 @@
+
```
Comment (Connoropolous) [inline crates/hdk/ON-WASM.md:1]: You can ignore this files "changes"... I didn't change anything just copied it verbatim from the original.

File: crates/hdk/README.md [hunk 2]
```diff
@@ -310,554 +284,4 @@ documentation.
 - Serde messagepack: https://github.com/3Hren/msgpack-rust
 - Holochain serialized bytes: https://github.com/holochain/holochain-serialization/tree/develop/crates/holochain_serialized_bytes
 
-
```
Comment (Connoropolous) [inline crates/hdk/README.md:313]: All this text simply went verbatim into "ON-WASM.md"

File: crates/hdk/README.md [hunk 3]
```diff
@@ -219,15 +193,15 @@ Holochain exposes a list of holochain-specific things that a wasm can do.
 For detailed documentation of the full list, see the `core/ribosome` module
 inside core, but some illustrative examples include:
 
-- `emit_signal`: publish data to subscribed clients
-- `encrypt` & `decrypt`: use the agent's keypair to encrypt and decrypt data
```
Comment (Connoropolous) [inline crates/hdk/README.md:223]: these don't exist in this form anymore...

File: crates/hdk/README.md [hunk 4]
```diff
@@ -101,16 +76,16 @@ debug!(links);
...
 let zome_name = zome_info()?.zome_name;
-// Call your friends foo function
+// Call your friends "foo" function
 let result: SerializedBytes = call_remote(
     my_friends_agent_pubkey,
     zome_name,
     "foo".to_string(),
     CapSecret::default(),
-    MyInput.try_into()?
+    MyInput
```
Comment (Connoropolous) [inline crates/hdk/README.md:85]: MyInput just needs to be serializable here, not serialized, so that's why it's fine to pass this

**Review by Connoropolous — COMMENTED** [review]
just adding context

File: crates/hdk/README.md [hunk 5]
```diff
@@ -1,20 +1,9 @@
 # Holochain Development Kit (HDK)
 
-This HDK is currently in flux, expect rapid changes.
-
-There are a few big differences between this kit and previous kits.
```
Comment (pdaoust) [inline crates/hdk/README.md:5]: This feels like it's a necessary preamble to the following bullet points.

File: crates/hdk/README.md [hunk 6]
```diff
@@ -219,15 +193,15 @@ Holochain exposes a list of holochain-specific things that a wasm can do.
...
 - `sign`: use the agent's keypair to sign some data
-- `commit_entry`: save some data to the local source chain and broadcast it to
+- `create_entry`: save some data to the local source chain and broadcast it to
   the DHT to be redundantly validated and stored
 - `get_entry`: retrieve some data from local or the network given its hash
 - `create_link`: create graph style relationships (links) between entries
 - `get_links`: retrive links between entries using the DHT as a graph database
-- `send`: send data directly to a known peer on the network
+- `remote_signal`: send data directly to known peers on the network without waiting for a response
+- `remote_call`: perform a remote procedure call on a peers node, if you're authorized
```
Comment (pdaoust) [inline crates/hdk/README.md:204]: This one should be `call_remote`, which maybe indicates that `remote_signal` should be renamed to `emit_signal_remote` to make it similar to both `call_remote` and `emit_signal` in the next HDK sweep?

File: crates/hdk/README.md (hunk 3 above)
Comment (pdaoust) [reply crates/hdk/README.md:223]: `emit_signal` still does: https://docs.rs/hdk/0.0.100/hdk/p2p/fn.emit_signal.html

**Review by pdaoust — CHANGES_REQUESTED** [review]
THANK YOU SO MUCH FOR THIS @Connoropolous . This was on my list of "things I know need to be updated". Just a few errata to note.

File: crates/hdk/README.md (hunk 6 above)
Comment (guillemcordoba) [reply crates/hdk/README.md:204]: I would agree with this :D

File: crates/hdk/README.md (hunk 3 above)
Comment (Connoropolous) [reply crates/hdk/README.md:223]: oops, meant for only encrypt decrypt to come out

File: crates/hdk/README.md (hunk 5 above)
Comment (Connoropolous) [reply crates/hdk/README.md:5]: I hear you, I think having the very first thing you read being a contrast with a previous way of doing things, is pretty unhelpful. Best to expand / reframe the thing as its intended to act and function now?

File: crates/hdk/README.md (hunk 3 above)
Comment (Connoropolous) [reply crates/hdk/README.md:223]: Oh actually it's still there, it's just down below this comment block. Tweaked that text.

File: crates/hdk/README.md (hunk 6 above)
Comment (Connoropolous) [reply crates/hdk/README.md:204]: yeah that'd be a nice change. I will update the call_remote

Comment (Connoropolous) [reply crates/hdk/README.md]: can you capture that notion somewhere else @pdaoust and I'll resolve this conversation

File: crates/hdk/README.md (hunk 5 above)
Comment (Connoropolous) [reply crates/hdk/README.md:5]: @pdaoust what's your take on this one? do you want to propose something to resolve it, do you want me to try to write something instead? can we just skip it and merge?

Comment (pdaoust) [reply crates/hdk/README.md:5]: @Connoropolous Yeah, I hear what you're saying about leading with "this thing is better than that old thing you never used. Re-reading, I think it's just fine, actually. 👍

File: crates/hdk/README.md (hunk 6 above)
Comment (pdaoust) [reply crates/hdk/README.md]: Happy to; I'll create a PR in fact.

**Review by pdaoust — APPROVED** [review]

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**